### PR TITLE
[REVIEW] Client Side Workflow (shell)

### PR DESCRIPTION
**Pull request by calvinjia**
A portion of #2523 to have all `shell` files available for review.

### CODE REVIEW

File: shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java [hunk 1]
```diff
@@ -46,40 +45,40 @@ protected AbstractLsCommand(TachyonConf conf, TachyonFileSystem tfs) {
...
           inMemory = "Not In Memory";
         }
       }
       System.out.format(Constants.COMMAND_FORMAT_LS,
-          FormatUtils.formatPermission((short) file.getPermission(), file.isFolder),
-          file.getUserName(), file.getGroupName(), FormatUtils.getSizeFromBytes(file.getLength()),
-          CommandUtils.convertMsToDate(file.getCreationTimeMs()), inMemory, file.getPath());
-      if (recursive && file.isFolder) {
-        ls(new TachyonURI(path.getScheme(), path.getAuthority(), file.getPath()), true);
+          CommandUtils.formatPermission(status.getPermission(), status.isFolder()),
```
Comment (apc999) [inline shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java]: `FormatUtils`?

File: shell/src/main/java/tachyon/shell/command/CommandUtils.java [hunk 2]
```diff
@@ -68,19 +66,54 @@ public static String convertMsToDate(long millis) {
   }
 
   /**
+   * Converts an int permission value to a formatted String.
+   *
+   * @param permission value of permission for the path
+   * @param isDir whether the path is a directory
+   * @return formatted permission String
+   */
+  public static String formatPermission(int permission, boolean isDir) {
```
Comment (apc999) [inline shell/src/main/java/tachyon/shell/command/CommandUtils.java]: why this method shows up here?

**Comment by apc999** [comment]
finished my pass on this PR. left two comments.

File: shell/src/main/java/tachyon/shell/command/LoadCommand.java [hunk 3]
```diff
@@ -65,25 +63,22 @@ void runCommand(TachyonURI path) throws IOException {
    */
   private void load(TachyonURI filePath) throws IOException {
     try {
-      TachyonFile fd = mTfs.open(filePath);
-      FileInfo fInfo = mTfs.getInfo(fd);
-      if (fInfo.isFolder) {
-        List<FileInfo> files = mTfs.listStatus(fd);
-        Collections.sort(files);
```
Comment (gpang) [inline shell/src/main/java/tachyon/shell/command/LoadCommand.java]: sort no longer necessary?

**Comment by gpang** [comment]
Just a minor comment. Thanks!

**Comment by aaudiber** [comment]
LGTM aside from bin's comments

File: shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java [hunk 4]
```diff
@@ -34,7 +33,7 @@
  */
 public abstract class AbstractLsCommand extends WithWildCardPathCommand {
 
-  protected AbstractLsCommand(TachyonConf conf, TachyonFileSystem tfs) {
+  protected AbstractLsCommand(TachyonConf conf, FileSystem tfs) {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java]: please also rename the variable

File: shell/src/main/java/tachyon/shell/command/AbstractTfsShellCommand.java [hunk 5]
```diff
@@ -15,19 +15,19 @@
...
 /**
  * The base class for all the {@link TfsShellCommand} classes. It provides a default argument
- * validation method and a place to hold the {@link TachyonFileSystem} client.
+ * validation method and a place to hold the {@link FileSystem} client.
  */
 public abstract class AbstractTfsShellCommand implements TfsShellCommand {
 
   protected TachyonConf mTachyonConf;
-  protected TachyonFileSystem mTfs;
+  protected FileSystem mTfs;
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/AbstractTfsShellCommand.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/AbstractTfsShellCommand.java [hunk 6]
```diff
@@ -15,19 +15,19 @@
...
 
-  protected AbstractTfsShellCommand(TachyonConf conf, TachyonFileSystem tfs) {
+  protected AbstractTfsShellCommand(TachyonConf conf, FileSystem tfs) {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/AbstractTfsShellCommand.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CatCommand.java [hunk 7]
```diff
@@ -37,7 +36,7 @@
    * @param conf the configuration for Tachyon
    * @param tfs the filesystem of Tachyon
    */
-  public CatCommand(TachyonConf conf, TachyonFileSystem tfs) {
+  public CatCommand(TachyonConf conf, FileSystem tfs) {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CatCommand.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CatCommand.java [hunk 8]
```diff
@@ -49,13 +48,11 @@ public String getCommandName() {
   @Override
   void runCommand(TachyonURI path) throws IOException {
     try {
-      TachyonFile fd = mTfs.open(path);
-      FileInfo tFile = mTfs.getInfo(fd);
+      URIStatus status = mTfs.getStatus(path);
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CatCommand.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CommandUtils.java [hunk 9]
```diff
@@ -46,11 +45,10 @@ private CommandUtils() {
    *        whether the file is pinned; {@link Constants#NO_TTL} means to unset the TTL value
    * @throws IOException when failing to set/unset the TTL
    */
-  public static void setTtl(TachyonFileSystem tfs, TachyonURI path, long ttlMs) throws IOException {
+  public static void setTtl(FileSystem tfs, TachyonURI path, long ttlMs) throws IOException {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CommandUtils.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CommandUtils.java (hunk 2 above)
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CommandUtils.java]: I think you wrongly kept this from merging conflicts.

File: shell/src/main/java/tachyon/shell/command/CommandUtils.java [hunk 10]
```diff
@@ -68,19 +66,54 @@ public static String convertMsToDate(long millis) {
...
    * Sets pin state for the input path
    *
-   * @param tfs The {@link TachyonFileSystem} client
+   * @param tfs The {@link FileSystem} client
    * @param path The {@link TachyonURI} path as the input of the command
    * @param pinned the state to be set
    * @throws IOException if a non-Tachyon related exception occurs
    */
-  public static void setPinned(TachyonFileSystem tfs, TachyonURI path, boolean pinned)
+  public static void setPinned(FileSystem tfs, TachyonURI path, boolean pinned)
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CommandUtils.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CopyFromLocalCommand.java [hunk 11]
```diff
@@ -50,7 +48,7 @@
    * @param conf the configuration for Tachyon
    * @param tfs the filesystem of Tachyon
    */
-  public CopyFromLocalCommand(TachyonConf conf, TachyonFileSystem tfs) {
+  public CopyFromLocalCommand(TachyonConf conf, FileSystem tfs) {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CopyFromLocalCommand.java]: rename the variable

File: shell/src/main/java/tachyon/shell/command/CopyToLocalCommand.java [hunk 12]
```diff
@@ -46,7 +45,7 @@
    * @param conf the configuration for Tachyon
    * @param tfs the filesystem of Tachyon
    */
-  public CopyToLocalCommand(TachyonConf conf, TachyonFileSystem tfs) {
+  public CopyToLocalCommand(TachyonConf conf, FileSystem tfs) {
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/command/CopyToLocalCommand.java]: rename the variable and for the rest of this PR

File: shell/src/main/java/tachyon/shell/command/CommandUtils.java (hunk 2 above)
Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/command/CommandUtils.java]: Good catch, I've removed the additional method.

File: shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java (hunk 1 above)
Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java]: Updated, thanks.

File: shell/src/main/java/tachyon/shell/command/LoadCommand.java (hunk 3 above)
Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/command/LoadCommand.java]: The previous sort on `FileInfo` wasn't a logical sort, so I don't think its necessary.

**Comment by gpang** [comment]
LGTM

File: shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java (hunk 4 above)
Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java]: Do you mind if we leave the variable renaming for another PR?

Comment (jsimsa) [inline shell/src/main/java/tachyon/shell/command/AbstractLsCommand.java]: OOC, how do you plan to identify the variables that should be renamed?

**Comment by jsimsa** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM

**Comment by apc999** [comment]
LGTM

**Comment by apc999** [comment]
Parent PR #2523 merged. Close this PR
